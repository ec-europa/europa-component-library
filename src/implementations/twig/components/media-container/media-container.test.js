import {
  merge,
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from '@ecl/test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';

import demoContentImg from '@ecl/specs-component-media-container/demo/data--image';
import demoContentVideo from '@ecl/specs-component-media-container/demo/data--video';
import demoBackwardVideo from '@ecl/specs-component-media-container/demo/old-data--video';
import demoContentEmbed from '@ecl/specs-component-media-container/demo/data--embed-video';
import demoContentInfographic from '@ecl/specs-component-media-container/demo/data--infographic';

expect.extend(toHaveNoViolations);

describe('Media Container', () => {
  const template = '@ecl/media-container/media-container.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);
  const defaultDataStructure = demoContentImg;

  describe('generic tests', () => {
    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(defaultDataStructure, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(defaultDataStructure, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(
          await renderTwigFileAsHtml(template, defaultDataStructure, true),
        ),
      ).toHaveNoViolations();
    });
  });

  describe('image', () => {
    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(defaultDataStructure)).resolves.toMatchSnapshot();
    });
  });

  describe('video', () => {
    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(demoContentVideo)).resolves.toMatchSnapshot();
    });

    test('as autoplay renders correctly', () => {
      const dataVideo = {
        ...demoContentVideo.video,
        autoplay: true,
      };

      expect.assertions(1);
      return expect(render(dataVideo)).resolves.toMatchSnapshot();
    });

    test('with old data renders correctly', () => {
      expect.assertions(1);
      return expect(render(demoBackwardVideo)).resolves.toMatchSnapshot();
    });

    test('with embedded media renders correctly', () => {
      expect.assertions(1);
      return expect(render(demoContentEmbed)).resolves.toMatchSnapshot();
    });
  });

  describe('infographic', () => {
    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(demoContentInfographic)).resolves.toMatchSnapshot();
    });
  });
});
