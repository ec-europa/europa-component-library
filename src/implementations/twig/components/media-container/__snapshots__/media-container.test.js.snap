// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC Media Container EC - Media container image Media container image renders correctly 1`] = `
<figure
  class="ecl-media-container"
>
  <img
    alt="Jean Monnet banner"
    class="ecl-media-container__media"
    src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
  />
  <figcaption
    class="ecl-media-container__caption"
  >
    The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
  </figcaption>
</figure>
`;

exports[`EC Media Container EC - Media container video renders correctly 1`] = `
<figure
  class="ecl-media-container"
>
  <video
    class="ecl-media-container__media"
    controls=""
    poster="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
  >
    <source
      src="https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/720/Big_Buck_Bunny_720_10s_2MB.mp4"
      type="video/mp4"
    />
    <source
      src="https://test-videos.co.uk/vids/bigbuckbunny/webm/vp9/720/Big_Buck_Bunny_720_10s_2MB.webm"
      type="video/webm"
    />
    <track
      kind="captions"
      label="English"
      src="/captions/bunny-en.vtt"
      srclang="en"
    />
    <track
      kind="captions"
      label="franÃ§ais"
      src="/captions/bunny-fr.vtt"
      srclang="fr"
    />
  </video>
  <figcaption
    class="ecl-media-container__caption"
  >
    The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
  </figcaption>
</figure>
`;

exports[`EC Media Container EC - Media container video with embedded media renders correctly 1`] = `
<figure
  class="ecl-media-container"
>
  <div
    class="ecl-media-container__media ecl-media-container__media--ratio-16-9"
  >
    <iframe
      allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
      allowfullscreen=""
      frameborder="0"
      height="197"
      src="https://www.youtube.com/embed/fgi-GSCB6ho"
      title="New digital strategy"
      width="350"
    />
  </div>
  <figcaption
    class="ecl-media-container__caption"
  >
    The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
  </figcaption>
</figure>
`;

exports[`EC Media Container Media Container generic tests Media container renders correctly with extra attributes 1`] = `
<figure
  class="ecl-media-container"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <img
    alt="Jean Monnet banner"
    class="ecl-media-container__media"
    src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
  />
  <figcaption
    class="ecl-media-container__caption"
  >
    The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
  </figcaption>
</figure>
`;

exports[`EC Media Container Media Container generic tests renders correctly with extra class names 1`] = `
<figure
  class="ecl-media-container custom-class custom-class--test"
>
  <img
    alt="Jean Monnet banner"
    class="ecl-media-container__media"
    src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
  />
  <figcaption
    class="ecl-media-container__caption"
  >
    The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.
  </figcaption>
</figure>
`;
